### PR TITLE
Added snaps error notification

### PR DESCRIPTION
**Pull request by shanejonas**
This adds the error notification using the `HomeNotification` component. It also adds the `HomeNotification` to storybook.

Fixes: #11897 
  
image

### CODE REVIEW

File: ui/pages/home/home.container.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ import {
   setRecoveryPhraseReminderHasBeenShown,
   setRecoveryPhraseReminderLastShown,
   setNewNetworkAdded,
+  removeSnapError,
```
Comment (rekmarks) [inline ui/pages/home/home.container.js:33]: This should also be fenced.

File: ui/store/actions.js [hunk 2]
```diff
@@ -808,6 +808,15 @@ export function txError(err) {
...
 
+export function removeSnapError(msgData) {
+  return (dispatch) => {
+    return promisifiedBackground
+      .removeSnapError(msgData)
+      .then(() => updateMetamaskStateFromBackground())
+      .then((newState) => dispatch(updateMetamaskState(newState)));
+  };
+}
+
```
Comment (rekmarks) [inline ui/store/actions.js:819]: This should also be fenced.

File: ui/pages/home/home.component.js [hunk 3]
```diff
@@ -205,12 +212,48 @@ export default class Home extends PureComponent {
       setWeb3ShimUsageAlertDismissed,
       originOfCurrentTab,
       disableWeb3ShimUsageAlert,
+      removeSnapError,
```
Comment (rekmarks) [inline ui/pages/home/home.component.js:215]: The Snaps-specific imports and functionality should be fenced. See other UI files with fences for examples, e.g. here #12745.

**Review by rekmarks — CHANGES_REQUESTED** [review]
Some fencing-related issues to fix!

File: ui/components/app/home-notification/home-notification.stories.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import HomeNotification from './home-notification.component';
+
+export default {
+  title: 'UI/HomeNotification',
```
Comment (georgewrmarshall) [inline ui/components/app/home-notification/home-notification.stories.js:6]: Title should follow folder structure. I have a PR open that should update all the stories to follow this convention still trying to get it in 😅 
```suggestion
  title: 'Components/App/HomeNotification',
```

File: ui/components/app/home-notification/home-notification.stories.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  },
+};
+
+const noop = () => {
+  // noop
+};
+
+export const DefaultStory = (args) => (
+  <HomeNotification {...args} onAccept={noop} onIgnore={noop} />
+);
```
Comment (georgewrmarshall) [inline ui/components/app/home-notification/home-notification.stories.js:35]: You can use actions here and they will show up in the actions tab when you click on the accept or ignore buttons

```suggestion
    onAccept: {
      action: 'onAccept',
    },
    onIgnore: {
      action: 'onIgnore',
    },
    },
  },
};

export const DefaultStory = (args) => (
  <HomeNotification {...args} />
);
```

Screen Shot 2021-11-25 at 2 16 23 PM

File: ui/pages/home/index.scss [hunk 6]
```diff
@@ -152,4 +152,8 @@
     background: none;
     margin-left: 20px;
   }
+
+  &__error-message {
+    left: 10px;
```
Comment (georgewrmarshall) [inline ui/pages/home/index.scss:157]: Adhere to [8px grid](https://consensys.slack.com/archives/G1L7H42BT/p1637778098226300)
```suggestion
    left: 8px;
```

File: ui/components/app/home-notification/home-notification.stories.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+DefaultStory.storyName = 'Default';
+
+DefaultStory.args = {
+  ignoreText: 'foo',
```
Comment (georgewrmarshall) [inline ui/components/app/home-notification/home-notification.stories.js:40]: It's nice to show the UI with any props that usually would be used.
```suggestion
    acceptText: 'Accept text',
  checkboxText: 'Checkbox text',
  checkboxTooltipText: 'Checkbox tooltip text',
  ignoreText: 'Ignore text',
  infoText: 'Info text',
  descriptionText: 'Description text',
```

**Review by georgewrmarshall — COMMENTED** [review]
Nice work adding the story!! Made some small suggestions

File: ui/pages/home/index.scss (hunk 6 above)
Comment (shanejonas) [reply ui/pages/home/index.scss:157]: the other sides end up being 10 from here:
https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/multiple-notifications/index.scss#L59-L60

image

File: ui/components/app/home-notification/home-notification.stories.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+export const DefaultStory = (args) => (
+  <HomeNotification {...args} />
+);
+
+DefaultStory.storyName = 'Default';
+
+DefaultStory.args = {
+    acceptText: 'Accept text',
```
Comment (shanejonas) [inline ui/components/app/home-notification/home-notification.stories.js:42]:
```suggestion
  acceptText: 'Accept text',
```

File: ui/pages/home/index.scss (hunk 6 above)
Comment (georgewrmarshall) [reply ui/pages/home/index.scss:157]: Ah ok. I would push for changing them all to be `8px` but if it's outside the scope of this PR keep it 10px to be consistent.

**Review by georgewrmarshall — DISMISSED** [review]
UI and story look good to me! 💯

**Review by darkwing — APPROVED** [review]
Looks awesome!

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**